main.py

```diff
@@ -231,6 +230,31 @@ def kcidb_send_notification(data, context):
     message = SPOOL_CLIENT.pick(notification_id)
     if not message:
         return
+    # send message via email
+    send_email_notification(message, notification_id)
+    # Acknowledge notification as sent
+    SPOOL_CLIENT.ack(notification_id)
+
+
+def kcidb_pick_notifications():
+    """
+    Pick abandoned notifications and resend them
+    """
+    for notification_id in SPOOL_CLIENT.unpicked():
+        # Pick abandoned notification and resend
+        message = SPOOL_CLIENT.pick(notification_id)
+        if not message:
+            continue
+        # send message via email
+        send_email_notification(message, notification_id)
+        # Acknowledge notification as sent
+        SPOOL_CLIENT.ack(notification_id)
+
+
+def send_email_notification(message, notification_id):
```

Comment:
I think it's not worth making this function know about the notification and notification ID just to log a message about it. It's really not necessary for what it has to do. If we remove the `notification_id` argument, and just log that message right before each place where we call this function, then this function would be come simpler, more reusable, and less entangled with the rest of the code. We can even remove the word `notification` from its name, and just call it `send_email`, or even better, `send_message`, matching the way we call email messages throughout the code. Do you think you can do that?